plotly/presentation_objs/presentation_objs.py

```diff
@@ -0,0 +1,1199 @@
+
+    child = {
+        'children': children_list,
+        'id': id,
+        'props': props,
+        'type': boxtype
+    }
+
+    if boxtype == 'Text':
+        child['defaultHeight'] = 36
+        child['defaultWidth'] = 52
+        child['resizeVertical'] = False
+    if boxtype == 'CodePane':
+        child['defaultText'] = 'Code'
+
+    return child
+
+
+def _percentage_to_pixel(value, side):
+    if side == 'left':
+        return WIDTH * (0.01 * value)
+    elif side == 'top':
+        return HEIGHT * (0.01 * value)
+    elif side == 'height':
+        return HEIGHT * (0.01 * value)
+    elif side == 'width':
+        return WIDTH * (0.01 * value)
+
+
+def _return_box_position(left, top, height, width):
+    values_dict = {
+        'left': left,
+        'top': top,
+        'height': height,
+        'width': width,
+    }
+    for key in values_dict.keys():
+        if isinstance(values_dict[key], str):
+            if values_dict[key][-2:] != 'px':
+                raise exceptions.PlotlyError(
+                    NEEDED_STYLE_ERROR_MESSAGE
+                )
+            try:
+                var = float(values_dict[key][: -2])
+            except ValueError:
+                raise exceptions.PlotlyError(
+                    NEEDED_STYLE_ERROR_MESSAGE
+                )
+
+        else:
+            var = _percentage_to_pixel(values_dict[key], key)
+        values_dict[key] = var
+
+    return (values_dict['left'], values_dict['top'],
+            values_dict['height'], values_dict['width'])
+
+
+def _remove_extra_whitespace_from_line(line):
+    while line.startswith('\n') or line.startswith(' '):
+        line = line[1:]
+    while line.endswith('\n') or line.endswith(' '):
+        line = line[: -1]
+    return line
+
+
+def _list_of_slides(markdown_string):
+    if not markdown_string.endswith('\n---\n'):
+        markdown_string += '\n---\n'
+
+    text_blocks = re.split(
+        '\n--\n|\n---\n|\n----\n|\n-----\n|\n------\n', markdown_string
+    )
+    list_of_slides = []
+    for j, text in enumerate(text_blocks):
+        if not all(char in ['\n', '-', ' '] for char in text):
+            list_of_slides.append(text)
+
+    if '\n-\n' in markdown_string:
+        msg = ("You have at least one '-' by itself on its own line in your "
+               "markdown string. If you are trying to denote a new slide, "
+               "make sure that the line has 3 '-'s like this: \n\n---\n\n"
+               "A new slide will NOT be created here.")
+        warnings.warn(msg)
+
+    return list_of_slides
+
+
+def _boxes_in_slide(slide):
+    boxes = []
+    slide_copy = copy.deepcopy(slide)
+    prop_split = ';'
+    prop_val_sep = '='
+
+    while '.left' in slide_copy:
+        prop_dict = {}
+        left_idx = slide_copy.find('.left')
+        l_brace_idx = slide_copy[left_idx:].find('{{') + left_idx
+        properties = slide_copy[left_idx + 1: l_brace_idx].split(
+            prop_split
+        )
```

Comment:
could you add some comments that describe what this function and this loop is doing?

Reply:
this is a unused function - deleting